src/autocomplete.js

```diff
@@ -490,6 +487,36 @@ export default class Autocomplete extends Component {
       inputClassList.push(inputClasses)
     }
 
+    const menuClassName = `${cssNamespace}__menu`
+    const menuModifierDisplayMenu = `${menuClassName}--${displayMenu}`
+    const menuIsVisible = menuOpen || showNoOptionsFound
+    const menuModifierVisibility = `${menuClassName}--${(menuIsVisible) ? 'visible' : 'hidden'}`
+
+    const safeMenuAttributes = sanitiseMenuAttributes(menuAttributes || {})
```

Comment:
Instead of lots of new sanitisation code, can we overwrite role/id instead?

```js
menuAttributes.id = id
menuAttributes.role = role
```

Reply:
That's a good idea for avoiding to `delete` the `id` and the `role`, didn't think of that. We could do the same with `class` and `className` (we'd need to set both, but I don't think that's a massive issue. Could try setting one to `null` depending on how things work with React/Preact). 

I'd still be keen to clone the object to avoid modifying the prop we're given, or do you think it's over the top?

Reply:
With object spread?

We had `{...menuAttributes}` on line 538 previously:

https://github.com/alphagov/accessible-autocomplete/blob/40350ae89328e813cc9437e4aac8d52ef813dd4a/src/autocomplete.js#L538C11-L538C30

Are you thinking to overwrite role/id with `{...menuAttributes, id, role}` instead?

Reply:
Yeah, just with an object spread, so that when we set the `id` or `role`, it doesn't leak the modifications to the object that was passed as props directly. Thinking this'll end up with something like:

```js
// Compute the menu classes from the array
// including `menuAttributes.class` and `menuAttributes.className`
const menuClasses = ...

const computedMenuAttributes = {
  ...menuAttributes,
  // Override the attributes the component computes
  id: `${id}__listbox`,
  role: 'listbox',
  className: menuClasses.join(' ')
}

// Figure a way to delete the `class` property
delete computedMenuAttributes.class

// Then in the JSX, leaving the handler here for consistency with the other elements
// whose handlers are set  in JSX. But could equaly go in `computedMenuAttributes` as well.
<ul 
	onMouseLeave={(event) => this.handleListMouseLeave(event)}
	{...computedMenuAttributes} >
```